### PR TITLE
Upgrade package skopeo to 0.14.2, to take Docker Daemon fix

### PR DESCRIPTION
**Pull request by dsteeley**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
See https://github.com/containers/skopeo/compare/v1.14.1...v1.14.2 for the details of the update. 
0.14.2 fixes version compatibility issues with docker 25.0.0 onwards. 

###### Change Log  

- Update skopeo to 0.14.2 resolving Docker Daemon issue
- Change
- Change

###### Does this affect the toolchain?  
**NO**

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

**Comment by dsteeley** [comment]
Please could I request that the skopeo 0.14.2 tarball be uploaded from https://github.com/containers/skopeo/releases/tag/v1.14.2 to `https://cblmarinerstorage.blob.core.windows.net/sources/core/`. 
Once this is uploaded I should be able to test the build and checks.

File: SPECS/skopeo/skopeo.spec
Comment (mfrw) [inline SPECS/skopeo/skopeo.spec:49]: Test build: [PR-7859](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=504823&view=results)

Comment (dsteeley) [reply SPECS/skopeo/skopeo.spec:49]: @mfrw that pipeline has passed. Do you require anything more from me to get this PR merged?

Comment (mfrw) [reply SPECS/skopeo/skopeo.spec:49]: Let's wait for a review from one of the stable maintainers.
/cc @jslobodzian @christopherco

Comment (dsteeley) [reply SPECS/skopeo/skopeo.spec:49]: Bump on this @jslobodzian/ @christopherco. Could you please review?

Comment (RocketJas) [reply SPECS/skopeo/skopeo.spec:49]: Is is this MR ready to go  @jslobodzian / @christopherco ?

**Review by mfrw — APPROVED** [review]
LGTM

**Pull request merged**
